lib/crystalball/execution_map.rb

```diff
@@ -30,7 +36,19 @@ def to_h
     # @param [Hash] initial list of cases
     def initialize(metadata: {}, cases: {})
       @cases = cases
-      @metadata = Metadata.new(type: self.class.name, **metadata)
+
+      version = metadata[:version].to_f
+
+      if cases.any?
+        version = 1.0 if version.zero?
```

Comment:
This does exactly what `guard_version_compatibility` does.

Reply:
So this could be done after [this](https://github.com/toptal/crystalball/pull/44/files#diff-5fc912a5f464543e946a3414e981f7f7R40) and avoid repeting